File: ios/Theme/Button.swift
import Foundation

public class Button: ViewStyle {
  private static let KEY: String = "guidance"
  private let target: NSDictionary?
  private let font: Font
  private let color: Color

  init(target: NSDictionary?) {
    self.target = Style().getTarget(target, key: Button.KEY)
    self.font = Font()
    self.color = Color()
    
    super.init(key: Button.KEY)
  }
  
  override public func getCornerRadius() -> Int32 {
    return super.getCornerRadius(-1)
  }
  
  public func getFont() -> UIFont {
    return self.font.getTypography(theme: self.target, key: "font")
  }

  public func getBackgroundNormalColor() -> UIColor {
    return self.color.getColor(self.target, key: "backgroundNormalColor", defaultColor: "#026ff4")
  }

  public func getBackgroundDisabledColor() -> UIColor {
    return self.color.getColor(self.target, key: "backgroundDisabledColor", defaultColor: "#b3d4fc")
  }

  public func getBackgroundHighlightColor() -> UIColor {
    return self.color.getColor(self.target, key: "backgroundHighlightColor", defaultColor: "#0264dc")
  }

  public func getTextNormalColor() -> UIColor {
    return self.color.getColor(self.target, key: "textNormalColor")
  }

  public func getTextDisabledColor() -> UIColor {
    return self.color.getColor(self.target, key: "textDisabledColor")
  }

  public func getTextHighlightColor() -> UIColor {
    return self.color.getColor(self.target, key: "textHighlightColor")
  }
}

File: examples/flutterapp/android/app/src/main/java/com/azifacemobile/theme/Button.java
package com.azifacemobile.theme;

import com.azifacemobile.utils.Theme;

import org.json.JSONObject;

public class Button {
  private final JSONObject theme;
  private final Color color;

  public Button(JSONObject theme) {
    this.theme = new Theme().getTarget(theme, "button");
    this.color = new Color();
  }

  public int getBackgroundNormalColor() {
    return this.color.getColor(this.theme, "backgroundNormalColor", "#026ff4");
  }

  public int getBackgroundDisabledColor() {
    return this.color.getColor(this.theme, "backgroundDisabledColor", "#b3d4fc");
  }

  public int getBackgroundHighlightColor() {
    return this.color.getColor(this.theme, "backgroundHighlightColor", "#0264dc");
  }

  public int getTextNormalColor() {
    return this.color.getColor(this.theme, "textNormalColor");
  }

  public int getTextDisabledColor() {
    return this.color.getColor(this.theme, "textDisabledColor");
  }

  public int getTextHighlightColor() {
    return this.color.getColor(this.theme, "textHighlightColor");
  }
}
